keps/0001a-meta-kep-implementation.md

```diff
@@ -0,0 +1,213 @@
+---
+kep-number: 1a
+title: Meta KEP Implementation
+authors:
+  - "@justaugustus"
+  - "@calebamiles"
+  - "@jdumars"
+owning-sig: sig-pm
+participating-sigs:
+  - sig-architecture
+reviewers:
+  - "@idvoretskyi"
+  - "@timothysc"
+approvers:
+  - "@bgrant0607"
+  - "@jbeda"
+editor:
+creation-date: 2018-09-08
+last-updated: 2018-09-18
+status: provisional
+see-also:
+  - KEP-1
+replaces:
+superseded-by:
+---
+
+# Meta KEP Implementation
+
+## Table of Contents
+
+- [Table of Contents](#table-of-contents)
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Why another KEP?](#why-another-kep)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [Implementation Details / Notes / Constraints](#implementation-details--notes--constraints)
+    - [Define](#define)
+    - [Organize](#organize)
+    - [Visibility and Automation](#visibility-and-automation)
+    - [Constraints](#constraints)
+- [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+
+## Summary
+
+Drive KEP adoption through improved process, documentation, visibility, and automation.
+
+## Motivation
+
+The KEP process is the standardized structure for proposing changes to the Kubernetes project.
+
+In order to graduate KEPs to GA, we must iterate over the implementation details.
+
+This KEP seeks to define actionable / delegable items to move the process forward.
+
+Finally, by submitting a KEP, we gain an opportunity to dogfood the process and further identify areas for improvement.
+
+### Why another KEP?
+
+As [KEP-1] is currently de facto for the project, we must be careful to make changes to it in an iterative and atomic fashion.
+
+When proposing a KEP, we action over some unit of work, usually some area of code.
+
+In this instance, we treat [KEP-1] as the unit of work. That said, this would be considered a meta-KEP of the meta-KEP.
+
+### Non-Goals
+
+- API Review process
+- Feature request triage
+- Developer guide
+
+## Proposal
+
+### Implementation Details / Notes / Constraints
+
+#### Define
+
+- Refine existing KEP documentation
+- Define KEP [DACI]
+- Glossary of terms (enhancement, KEP, feature, etc.)
+- KEP Workflow
+  - KEP states
+  - Entry / exit criteria
+
+#### Organize
+
+- Move KEPs from flat-files to a directory structure:
+```
+├── keps # top-level directory
+│   ├── sig-beard # SIG directory
```

Comment:
Why are KEPs by SIG?  This means people have to look in multiple places to find stuff.  Compare to python PEP-0 where you only have to look at one page to find all the KEPs.

Or the index of all the RFCs from IETF: 
https://tools.ietf.org/rfc/index

Reply:
I feel like repository layout and visual presentation shouldn't be conflated. The job of a KEP is to convey an intent, as well as track how the work to satisfy the intent evolves over time. With that lens, the layout is best oriented to serve the highest stakeholders which are subproject participants first and foremost, SIGs second, consumers like the release team third, and causal observers dead last. Focusing on the current visualization as a starting point feels like a red herring since the KEP process itself, state transition definitions, cli tooling, etc. is not formalized or implemented. Reuse of the OWNERS construct at the subproject level makes a ton of sense.

Reply:
I'm stuck on repo layout and such because ever since KEPs were first suggested we've been using the "automation will make this more digestible" line, yet the automation hasn't arrived. Until it does, this is the experience we're left with.  Imposing even more friction up front doesn't fill me with confidence that automation will come any sooner.